### PR TITLE
Success condition for precompiled contracts.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libethereum/Executive.cpp [hunk 1]
```diff
@@ -262,9 +262,9 @@ bool Executive::call(CallParameters const& _p, u256 const& _gasPrice, Address co
...
 	m_savepoint = m_s.savepoint();
 
-	if (m_sealEngine.isPrecompiled(_p.codeAddress))
+	if (m_sealEngine.isPrecompiled(_p.codeAddress, m_envInfo.number()))
 	{
-		bigint g = m_sealEngine.costOfPrecompiled(_p.codeAddress, _p.data);
+		bigint g = m_sealEngine.costOfPrecompiled(_p.codeAddress, _p.data, m_envInfo.number());
 		if (_p.gas < g)
 		{
 			m_excepted = TransactionException::OutOfGasBase;
```
Comment (chriseth) [inline libethereum/Executive.cpp]: @chfast do you know why this does not do `m_gas = 0`?

Comment (chfast) [inline libethereum/Executive.cpp]: Probably setting excepted flag is enough... As you can see we have probably too much layers here :).
Try setting it to 0. If no broken tests we can keep it.

**Comment by chriseth** [comment]
I used `pair<bool,bytes>` because of the upcoming EIP for the revert opcode.

**Comment by chfast** [comment]
How is this related?

**Comment by chriseth** [comment]
The revert opcode allows returning data in the failure case. So in the future, it looks like we have the following parameters:
 - failure or not
 - returned data
 - remaining gas (because we do not out-of-gas for all failure cases)

**Pull request merged**